ui/src/app/job-list/table/table.component.html

```diff
@@ -67,7 +67,43 @@
     <ng-container matColumnDef="{{ df.field }}">
       <mat-header-cell *matHeaderCellDef>{{ df.display }}</mat-header-cell>
       <mat-cell class="additional-column" *matCellDef="let j">
-        <div class="display-field" *ngIf=!isStatusField(df)>{{ getFieldValue(j, df) }}</div>
+        <div class="display-field" *ngIf=canEdit(df)>
+          <div [ngSwitch]="getFieldType(df)">
+            <div *ngSwitchCase="'text'">
+              <div class="edit-field">
+                <span class="field-value">{{ getFieldValue(j, df) }}</span>
+                <button class="jobs-dropdown-arrow" mat-icon-button [matMenuTriggerFor]="menu" [class.visible]="showDropdownArrow(j)">
+                  <img src="{{ getDropdownArrowUrl() }}">
+                </button>
+              </div>
+              <mat-menu #menu="matMenu" class="wide-menu" xPosition="before" yPosition="below" [overlapTrigger]="false">
+                <mat-form-field>
+                  <input matInput
+                         #labelField
+                         value="{{ getFieldValue(j, df) }}"
+                         maxlength="255"
+                         (click)="$event.stopPropagation()">
+                  <mat-hint align="end">{{labelField.value?.length || 0}}/255</mat-hint>
+                </mat-form-field>
+                <button class="apply-link" (click)="setFieldValue(j, df.field, labelField.value)">Apply</button>
+              </mat-menu>
+            </div>
+            <mat-form-field *ngSwitchCase="'list'">
+              <mat-select #labelSelect
+                          [value]="getFieldValue(j, df) ? getFieldValue(j, df) : null"
```

Comment:
Is the ternary necessary here? What falsey value are you guarding against that you want to convert to null?

Reply:
this is so that, if the field is empty, it won't select and display the "(none)" option, which muddies up the table

Reply:
I see, `|| null` would be slightly more readable in that case

Reply:
good point – done